llvm/lib/CodeGen/ReplaceWithVeclib.cpp

```diff
@@ -129,7 +143,7 @@ static bool replaceWithCallToVeclib(const TargetLibraryInfo &TLI,
   // using scalar argument types.
   std::string ScalarName =
       Intrinsic::isOverloaded(IID)
-          ? Intrinsic::getName(IID, ScalarArgTypes, II->getModule())
```

Comment:
I feel weird about this. Does `ScalarArgTypes` get used anywhere else? From the PR perspective now the code that pushes to the `SmallVector` isn't being used, but you have left it in.

Reply:
It's used to construct the function type `ScalarFTy` below.  This is why it needs all the argument types.